### PR TITLE
Added uplink port db for vlrouter

### PR DESCRIPTION
**Pull request by DivyaVavili**
PR includes
- Uplink DB for vlrouter mode
- Delete necessary flows during RemoveUplink for vlanBridge and vlrouter mode

### CODE REVIEW

File: vlanBridge.go [hunk 1]
```diff
@@ -577,6 +577,11 @@ func (vl *VlanBridge) RemoveUplink(uplinkName string) error {
 
 	// Stop monitoring links in the port
 	for _, link := range uplinkPort.MbrLinks {
+		// Uninstall the flow entry
+		vl.portVlanFlowDb[link.OfPort].Delete()
```
Comment (rchirakk) [inline vlanBridge.go]: verify  map lookup status for safety

File: vlrouter.go [hunk 2]
```diff
@@ -1010,10 +1022,21 @@ func (self *Vlrouter) AddUplink(uplinkPort *PortInfo) error {
...
+	self.uplinkPortDb.Set(uplinkPort.Name, uplinkPort)
 	return nil
 }
 
 func (self *Vlrouter) RemoveUplink(uplinkName string) error {
+	uplinkPort := self.GetUplink(uplinkName)
+
+	for _, link := range uplinkPort.MbrLinks {
+		// Uninstall the flow entry
+		self.portVlanFlowDb[link.OfPort].Delete()
```
Comment (rchirakk) [inline vlrouter.go]: verify  map lookup status for safety

File: vlanBridge.go [hunk 3]
```diff
@@ -575,8 +575,20 @@ func (vl *VlanBridge) AddUplink(uplinkPort *PortInfo) error {
 func (vl *VlanBridge) RemoveUplink(uplinkName string) error {
 	uplinkPort := vl.GetUplink(uplinkName)
 
+	if uplinkPort == nil {
+		return nil
```
Comment (jojimt) [inline vlanBridge.go]: How about an error log here?

File: vlrouter.go [hunk 4]
```diff
@@ -1010,10 +1022,28 @@ func (self *Vlrouter) AddUplink(uplinkPort *PortInfo) error {
...
+
+	self.uplinkPortDb.Set(uplinkPort.Name, uplinkPort)
 	return nil
 }
 
 func (self *Vlrouter) RemoveUplink(uplinkName string) error {
+	uplinkPort := self.GetUplink(uplinkName)
+
+	if uplinkPort == nil {
+		return nil
```
Comment (jojimt) [inline vlrouter.go]: Here too.

**Pull request merged**